### DISCUSSION TITLE
Rename from "Extra Care"

### DISCUSSION THREAD

**Discussion by drewhamilton** [drewhamilton/Poko#27] [Ideas]
I've received feedback that "Extra Care" may not be a great name for this library. It has a few pros, but (1) it does not per se suggest what the purpose of this library is, and (2) it is not especially memorable once you do know what the purpose this library is.

Changing names comes with its own set of cons, too. At Backbase we are using this plugin pretty widely now. Changing plugin ID means changing a bunch of build.gradle files. Worse, changing the package name would require changing dozens of source files. This could be mitigated with a slow deprecation process but I imagine would still be pretty annoying.

*If* I decide to change the name, there's a few options for a new name:
* Something boring but to the point, like "data-api-generator"
* "Poko", a play on the common terms "POJO" and "POCO"

I've also received the opposite feedback for the name in the past: "It's simple... Yet it's memorable, and suits the problem domain". So with that along with the annoyance of a migration I lean toward not changing. But opening this discussion to see if anyone else has strong opinions on this.

**Comment by ZacSweers** [comment]
I vote poko - easy to remember, you can kind of get an idea of what it does from the name, pun on poco (it does promise a smaller footprints), and for once it would be a kotlin library using the letter 'k' but unironically.

I've also forgotten the current name of this library when referencing it like 4 times, and could not find it in Google/GitHub search results each time either.

**Reply by drewhamilton** [reply]
Would you make the annotation `@Poko` too?

**Reply by ZacSweers** [reply]
I would make it configurable via the Gradle plugin like other compiler plugins do

**Reply by drewhamilton** [reply]
Yeah, I initially avoided that configurability but I think it's the right way to go. Filed #29.

**Reply by Hariofspades** [reply]
This library deserves Poko. At the same time as a colleague, I would hate to change every configuration we built :-D
I'm on board for minimal breaking changes

**Comment by ZacSweers** [comment]
![the 12 people have spoken](https://user-images.githubusercontent.com/1361086/104264517-cb072e00-5459-11eb-9747-e8ca5f849b13.png)

**Comment by drewhamilton** [comment]
All right, I've also received a couple private messages in favor of Poko, so I'm going to go forward with this. Need to think a bit about what the migration path ought to be, but [itshappening.gif].

**Reply by drewhamilton** [reply]
Opened #31
